src/polyswarm_api/endpoint.py

```diff
@@ -0,0 +1,254 @@
+        }
+
+    def lookup_uuid(self, uuid, **kwargs):
+        return {
+            'method': 'GET',
+            'url': '{}/uuid/{}'.format(self.community_base, uuid)
+        }
+
+    def _get_engine_names(self):
+        return {
+            'method': 'GET',
+            'url': '{}/microengines/list'.format(self.uri)
+        }
+
+    def submit_live_hunt(self, rule):
+        return {
+            'method': 'POST',
+            'url': '{}/live'.format(self.hunt_base),
+            'json': {'yara': rule.ruleset},
+        }
+
+    def live_lookup(self, with_bounty_results=True, with_metadata=True,
+                    limit=const.RESULT_CHUNK_SIZE, offset=0, id=None):
+        req = {
+            'method': 'GET',
+            'url': '{}/live/results'.format(self.hunt_base),
+            'params': {
+                'with_bounty_results': utils.bool_to_int[with_bounty_results],
+                'with_metadata': utils.bool_to_int[with_metadata],
+                'limit': limit,
+                'offset': offset,
+            },
+        }
+
+        if id:
+            req['params']['id'] = id
+
+        return req
+
+    def submit_historical_hunt(self, rule):
+        return {
+            'method': 'POST',
+            'url': '{}/historical'.format(self.hunt_base),
+            'json': {'yara': rule.ruleset},
+        }
+
+    def historical_lookup(self, with_bounty_results=True, with_metadata=True,
+                    limit=const.RESULT_CHUNK_SIZE, offset=0, id=None):
+        req = {
+            'method': 'GET',
+            'url': '{}/historical/results'.format(self.hunt_base),
+            'params': {
+                'with_bounty_results': utils.bool_to_int[with_bounty_results],
+                'with_metadata': utils.bool_to_int[with_metadata],
+                'limit': limit,
+                'offset': offset,
+            },
+        }
+
+        if id:
+            req['params']['id'] = id
+
+        return req
+
+    def historical_delete(self, hunt_id):
+        return {
+            'method': 'DELETE',
+            'url': '{}/historical'.format(self.hunt_base),
+            'json': {'hunt_id': hunt_id}
+        }
+
+    def live_delete(self, hunt_id):
+        return {
+            'method': 'DELETE',
+            'url': '{}/live'.format(self.hunt_base),
+            'json': {'hunt_id': hunt_id}
+        }
+
+    def historical_list(self):
+        return {
+            'method': 'GET',
+            'url': '{}/historical'.format(self.hunt_base),
+            'params': {'all': 'true'},
+        }
+
+    def live_list(self):
+        return {
+            'method': 'GET',
+            'url': '{}/live'.format(self.hunt_base),
+            'params': {'all': 'true'},
+        }
+
+
+class PolyswarmRequestExecutor(object):
+    """ This class accepts requests from a PolyswarmRequestGenerator and executes it """
+    def __init__(self, key, timeout=const.DEFAULT_HTTP_TIMEOUT, retries=const.DEFAULT_RETRIES, request_cls=None):
```

Comment:
I have made a point down below about getting instances rather than classes. The same applies here.